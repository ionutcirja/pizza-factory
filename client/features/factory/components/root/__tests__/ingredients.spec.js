import React from 'react';
import { shallow } from 'enzyme';
import Ingredients from '../ingredients';

describe('Ingredients component', () => {
  const propsToRender = {
    maxToppings: 2,
    basePrice: 10,
    toppings: [
      {
        defaultSelected: true,
        topping: {
          name: 'option 1',
          price: 1,
        },
      },
      {
        defaultSelected: false,
        topping: {
          name: 'option 2',
          price: 2,
        },
      },
      {
        defaultSelected: false,
        topping: {
          name: 'option 3',
          price: 3,
        },
      },
    ],
    setFieldValue: jest.fn(),
  };
  
  describe('render', () => {
    it('should render a checkbox group form field component', () => {
      const wrapper = shallow(<Ingredients {...propsToRender} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
  
  describe('componentDidMount', () => {
    it('should set base price field value', () => {
      shallow(<Ingredients {...propsToRender} />);
      expect(propsToRender.setFieldValue).toHaveBeenCalledWith('basePrice', propsToRender.basePrice);
    });
  });
  
  describe('componentDidUpdate', () => {
    it('should update base price field value if basePrice prop value is changed', () => {
      const wrapper = shallow(<Ingredients {...propsToRender} />);
      wrapper.instance().componentDidUpdate({ basePrice: 9 });
      expect(propsToRender.setFieldValue).toHaveBeenCalledWith('basePrice', 10);
    });
  });
});
