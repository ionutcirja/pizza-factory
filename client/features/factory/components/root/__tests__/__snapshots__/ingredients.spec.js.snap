// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredients component render should render a checkbox group form field component 1`] = `
<FieldWrapper>
  <FormField
    isFieldArray={true}
    label="Please select your desired toppings (maximum 2)"
    list={
      Array [
        Object {
          "name": "option 1",
          "value": 1,
        },
        Object {
          "name": "option 2",
          "value": 2,
        },
        Object {
          "name": "option 3",
          "value": 3,
        },
      ]
    }
    maxAllowed={2}
    name="toppings"
    type="checkbox_group"
  />
</FieldWrapper>
`;
